provers/mock/prover.go

```diff
@@ -39,20 +38,33 @@ func (pr *Prover) SetupForRelay(ctx context.Context) error {
 	return nil
 }
 
-// CreateMsgCreateClient creates a MsgCreateClient for the counterparty chain
-func (pr *Prover) CreateMsgCreateClient(selfHeader core.Header, signer sdk.AccAddress) (*clienttypes.MsgCreateClient, error) {
-	h := selfHeader.(*mocktypes.Header)
+// CreateInitialLightClientState creates a pair of ClientState and ConsensusState for building MsgCreateClient submitted to the counterparty chain
+func (pr *Prover) CreateInitialLightClientState(height exported.Height) (exported.ClientState, exported.ConsensusState, error) {
+	if height == nil {
+		if head, err := pr.GetLatestFinalizedHeader(); err != nil {
+			return nil, nil, fmt.Errorf("failed to get the latest finalized header: %v", err)
+		} else {
+			height = head.GetHeight()
+		}
+	}
```

Comment:
@siburu Should we check if `height` is less than the latest finalized height here?

Reply:
Thank you for pointing it out.
Yes, normal provers would fail to query for a finalized header at a height more than the latest finalized height, but the mock prover doesn't.
Therefore we should explicitly verify that the given height doesn't exceed the latest finalized height.

Reply:
@bluele I fixed it.